cartoframes/data/observatory/catalog/dataset.py

```diff
@@ -467,3 +472,16 @@ def subscription_info(self, credentials=None):
 
         return subscription_info.SubscriptionInfo(
             subscription_info.fetch_subscription_info(self.id, DATASET_TYPE, _credentials))
+
+    def _is_subscribed(self, credentials=None):
+        if self.is_public_data:
+            return True
+
+        _credentials = credentials or defaults.get_default_credentials()
+
+        if not isinstance(_credentials, Credentials):
+            raise ValueError('`credentials` must be a Credentials class instance')
+
+        datasets = Dataset.get_all({}, _credentials)
```

Comment:
Maybe it's more efficient here to use `Dataset.get(self.id)` instead of getting all the Datasets and check the item in the client.

Reply:
We are getting 'just' the datasets subscribed by the user. We do not have a getting only one method implement, but since we are going to filter by the user in the SQL, this is not going to be a problem in the db. And here in CARTOframes, I suppose the `self in datasets` is going to work over dozens of entities at most, so it will work fine too.

How do you see it?

Reply:
I thought that the `get` method was already implemented because of the `CatalogEntity` inheritance. However, it seems that it raises an exception if it doesn't get it. So between `get + try` or `get_all + in` we can keep the second option since the number of subscribed datasets is not huge.